src/theory/fp/fp_converter.cpp

```diff
@@ -12,25 +12,1029 @@
+								  (*mode).second,
+								  (*arg1).second,
+								  info.bvs,
+								  ubv(current[2])));
+	      i = u.find(current);
+	    }
+
+	    result = (*i).second.getNode();
+	  }
+	  break;
+
+	case kind::FLOATINGPOINT_TO_SBV_TOTAL :
+	  {
+	    TypeNode childType (current[1].getType());
+	    sbvMap::const_iterator i(s.find(current));
+
+	    if (i == s.end()) {
+	      rmMap::const_iterator mode(r.find(current[0]));
+	      fpMap::const_iterator arg1(f.find(current[1]));
+	      bool recurseNeeded = (mode == r.end()) || (arg1 == f.end());
+
+	      if (recurseNeeded) {
+		workStack.push(current);
+		if (mode == r.end()) { workStack.push(current[0]); }
+		if (arg1 == f.end()) { workStack.push(current[1]); }
+		continue;    // i.e. recurse!
+	      }
+
+	      FloatingPointToSBVTotal info =
+		current.getOperator().getConst<FloatingPointToSBVTotal>();
+
+
+	      s.insert(current, symfpu::convertFloatToSBV<traits>(fpt(childType),
+								  (*mode).second,
+								  (*arg1).second,
+								  info.bvs,
+								  sbv(current[2])));
+
+	      i = s.find(current);
+	    }
+
+	    result = (*i).second.getNode();
+	  }
+	  break;
+
+	case kind::FLOATINGPOINT_TO_UBV :
+	  Unreachable("Partially defined fp.to_ubv should have been removed by expandDefinition");
+	  break;
+
+	case kind::FLOATINGPOINT_TO_SBV :
+	  Unreachable("Partially defined fp.to_sbv should have been removed by expandDefinition");
+	  break;
+
+	  // Again, no action is needed
+	case kind::FLOATINGPOINT_COMPONENT_EXPONENT :
+	case kind::FLOATINGPOINT_COMPONENT_SIGNIFICAND :
+	case kind::ROUNDINGMODE_BITBLAST :
+	  /* Fall through ... */
+
+	default :
+	  PASSTHROUGH;
+	  break;
+	}
+
+      } else if (t.isReal()) {
+
+	switch (current.getKind()) {
+	  /******** Conversions ********/
+	case kind::FLOATINGPOINT_TO_REAL_TOTAL :
+	  {
+	    // We need to recurse so that any variables that are only
+	    // used under this will have components created
+	    // (via auxiliary constraints)
+
+	    TypeNode childType (current[0].getType());
+	    fpMap::const_iterator arg1(f.find(current[0]));
+
+	    if (arg1 == f.end()) {
+	      workStack.push(current);
+	      workStack.push(current[0]);
+	      continue;    // i.e. recurse!
+	    }
+
+	    // However we don't need to do anything explicitly with
+	    // this as it will be treated as an uninterpreted function
+	    // by the real theory and we don't need to bit-blast the
+	    // float expression unless we need to say something about
+	    // its value.
+	  }
+
+	  break;
+
+	case kind::FLOATINGPOINT_TO_REAL :
+	  Unreachable("Partially defined fp.to_real should have been removed by expandDefinition");
+	  break;
+
+	default :
+	  PASSTHROUGH;
+	  break;
+	}
+
+      } else {
+	PASSTHROUGH;
+      }
+    }
+
+    return result;
+  }
+
+
+  Node FpConverter::getValue (Valuation &val, TNode var) {
+    Assert(Theory::isLeafOf(var, THEORY_FP));
+
+    TypeNode t(var.getType());
+
+#ifdef REMOVE
```

Comment:
Is this a leftover for debugging purposes? If it can be removed, remove. This applies to all blocks guarded with `#ifdef REMOVE`.

Reply:
:-)